src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,187 @@
+        }
+
+        System.out.println(BORDER);
+    }
+
+    /**
+     * Prints a message to welcome the user.
+     */
+    public void printWelcomeMessage() {
+        System.out.println(BORDER);
+        System.out.println(" _________________________");
+        System.out.println("| Hello! I'm Duke         |");
+        System.out.println("| What can I do for you?  |");
+        System.out.println("| ________________________|");
+        System.out.println("|/");
+        System.out.println(BORDER);
+    }
+
+    /**
+     * Prints a message when the user inputs an incorrectly formatted command to add a Task.
+     * @param task Task which has the incorrect format.
+     */
+    public void printDescriptionErrorMessage(String task) {
+        System.out.println(BORDER);
+        System.out.println("Oops! The format of the " + task + " is incorrect!");
+        switch(task) {
+        case ("todo"):
+            System.out.println("Format for todo: todo DESCRIPTION");
+            break;
+        case ("event"):
+            System.out.println("Format for event: event DESCRIPTION /at dd-mm-yy hh:mm");
+            break;
+        case ("deadline"):
+            System.out.println("Format for deadline: deadline DESCRIPTION /by dd-mm-yy hh:mm");
+            break;
+        default:
+            break;
+        }
+        System.out.println(BORDER);
+    }
+
+    /**
+     * Prints a message when the user inputs a Task number that does not exist.
+     */
+    public void printTaskNumberErrorMessage() {
+        System.out.println(BORDER);
+        System.out.println("Oops! Please input a task number that exists!");
+        System.out.println(BORDER);
+    }
+
+    /**
+     * Prints a message when the user inputs an incorrectly formatted command to mark a Task as done.
+     */
+    public void printDoneErrorMessage() {
+        System.out.println(BORDER);
+        System.out.println("Oops! Please input a number after done!");
+        System.out.println(BORDER);
+    }
+
+    /**
+     * Prints a generic error message for all other errors.
+     */
+    public void printGenericErrorMessage() {
+        System.out.println(BORDER);
+        System.out.println("Oops! Something went wrong!");
+        System.out.println(BORDER);
+    }
+
+    /**
+     * Prints a message when the user inputs an incorrectly formatted date or time.
+     */
+    public void printDateTimeParseErrorMessage() {
+        System.out.println(BORDER);
+        System.out.println("Oops! Please make sure your date and time is in the format dd-mm-yy hh-mm!");
+        System.out.println(BORDER);
+    }
+
+    /**
+     * Prints an ArrayList of Tasks.
+     * @param tasks ArrayList to be printed.
+     */
+    public void printFilteredList(ArrayList<Task> tasks) {
+        System.out.println(BORDER);
+        if (tasks.size() == 0) {
+            System.out.println("There are no matching tasks!");
+        } else {
```

Comment:
Maybe consider [making the happy path prominent](https://nus-cs2113-ay2021s1.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent) here?